src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -163,6 +197,10 @@ public boolean isBootstrapRequired(Map<String, Object> newConfig)
         return nucleusConfigValidAndNeedsRestart || !bootstrapTaskStatusList.isEmpty();
     }
 
+    private boolean hasNotBeenCompleted(BootstrapTaskStatus task) {
```

Comment:
Here it was completed, just not necessarily successfully right? Maybe "incompleteOrErrored"?